Sources/StreamVideo/Utils/SerialActor/SerialActor.swift

```diff
@@ -4,61 +4,108 @@
-    ///     meaning it can be executed asynchronously. The block can return any value
-    ///     and can throw errors.
+    /// This method immediately cancels any tasks that are waiting to be
+    /// executed but have not yet started. Tasks that are currently running
+    /// will complete normally, but no new tasks will be processed.
     ///
-    /// - Throws: Any error thrown by the provided block.
-    /// - Returns: The value returned by the provided block.
-    func execute<T: Sendable>(_ block: @Sendable @escaping () async throws -> T) async throws -> T {
-        /// Create a new task that runs the provided block of code within a closure.
-        /// This closure captures the `previousTask` variable by value.
-        let task = Task<T, Error> { [previousTask] in
-            /// Wait for the previous task to finish by awaiting its result.
-            /// If the previous task is nil, this line does nothing.
-            _ = await previousTask?.result
-
-            try Task.checkCancellation()
-
-            /// Execute the provided block of code and re-throw any errors.
-            return try await block()
-        }
-        task.store(in: disposableBag)
-
-        /// Create a void task that we can store for synchronization
-        let voidTask = Task<Void, Error> {
-            _ = try await task.value
-        }
-        voidTask.store(in: disposableBag)
-
-        /// Update the `previousTask` variable to point to the void task.
-        previousTask = voidTask
+    /// This is a non-isolated method, meaning it can be called from any
+    /// context without requiring actor isolation.
+    nonisolated func cancel() {
+        serialExecutor.cancelAll()
+    }
 
-        try Task.checkCancellation()
-        
-        /// Wait for the newly created task to finish by awaiting its value.
-        /// This will re-throw any errors thrown by the block.
-        return try await task.value
+    /// Schedules a block to run serially on the actor's executor.
+    ///
+    /// - Parameter block: The asynchronous block to execute. Must be sendable
+    ///   and can throw errors.
+    /// - Returns: The result of the executed block.
+    /// - Throws: Rethrows any error thrown by the provided block.
+    ///
+    /// This method ensures that the provided block is executed serially with
+    /// respect to other tasks submitted to this actor. The block will be
+    /// queued and executed in the order it was submitted.
+    ///
+    /// ## Example
+    ///
+    /// ```swift
+    /// let result = try await serialActor.execute {
+    ///     // Perform some async work
+    ///     let data = try await fetchData()
+    ///     return processData(data)
+    /// }
+    /// ```
+    nonisolated func execute<T: Sendable>(
+        _ block: @Sendable @escaping () async throws -> T
+    ) async throws -> T {
+        let token = serialExecutor.registerJob()
+        return try await withTaskCancellationHandler(operation: {
```

Comment:
maybe I'm missing something here, but how do we ensure that the block is executed serially? Shouldn't we use the `serialExecutor` here?